server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -525,20 +522,19 @@ static void checkForIndexCompatibility(Logger logger, DataPath... dataPaths) thr
 
         if (metadata.oldestIndexVersion().isLegacyIndexVersion()) {
 
-            String bestDowngradeVersion = getBestDowngradeVersion(metadata.previousNodeVersion().toString());
             throw new IllegalStateException(
                 "Cannot start this node because it holds metadata for indices with version ["
-                    + metadata.oldestIndexVersion()
+                    + metadata.oldestIndexVersion().toReleaseVersion().split("-")[0]
```

Comment:
What's the split for? If the index version is genuinely between two release versions, we should display the range.

Reply:
I should have commented. What I'm seeing without the split is the message
```
Cannot start this node because it holds metadata for indices with version [1.52.39-7.0.0] with which this node of version [8.13.0] is incompatible. Revert this node to version [7.17.19-8.0.0] and delete any indices with versions earlier than [7.0.0] before upgrading to version [8.13.0]. If all such indices have already been deleted, revert this node to version [7.17.19-8.0.0] and wait for it to join the cluster to clean up any older indices from its metadata.
```

The range seems to appear because `toReleaseVersion` doesn't know about versions prior to 7.0.0, but the low end of the range would be informative. (In a real situation, I think we'd be seeing something like `6.8.2-7.0.0` but I haven't checked that specifically.)

Reply:
I think in that case it would make more sense for `ReleaseVersions` to just return the lowest range - as the ids are directly mappable pre ~8.9. We'll need to work out what to do when we move to ES 10 and drop all the v8 versions (maybe we keep them?), but for now we can do a direct conversion. I'll create a PR for that

Reply:
See #105335